### PR TITLE
Adding backward compatibility for sharding support in old DataLoader

### PR DESCRIPTION
**Pull request by VitalyFedyunin**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61312 Sort imports of test_datapipe.py
* **#61237 Adding backward compatibility for sharding support in old DataLoader**
* #61236 Implement usage of `is_shardable` and `apply_sharding`
* #61235 Add DataPipes Graph Functions
* #61234 [DataLoader] Adding demux and mux DataPipe-s

Differential Revision: [D29588832](https://our.internmc.facebook.com/intern/diff/D29588832)

### CODE REVIEW

File: test/test_datapipe.py [hunk 1]
```diff
@@ -1288,29 +1290,42 @@ def test_traverse_forked(self):
 
 
 class TestSharding(TestCase):
+
+    def get_pipeline(self):
```
Comment (ejguan) [inline test/test_datapipe.py:1294]: nit: Add an underscore in front of this function can prevent unittest over itself.

**Review by ejguan — APPROVED** [review]
LGTM.

**Comment by VitalyFedyunin** [comment]
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29588832).

**Comment by VitalyFedyunin** [comment]
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29588832).

**Pull request closed without merging**